addons/product_images/wizard/product_get_pic_using_barcode.py

```diff
@@ -0,0 +1,115 @@
+            raise UserError(_("You selected too many product variants at the same time. \
+Please select a maximum of 100 product variants in the same batch."))
+
+        default.update({
+            'product_ids': product_ids,
+            'nb_product': len(product_ids),
+        })
+
+        return default
+
+    product_ids = fields.Many2many(
+        comodel_name="product.product",
+        help="List of the selected product without a picture and with a barcode",
+    )
+    nb_product = fields.Integer(string="Number of product selected", readonly=True)
+
+    def get_pic_using_barcode(self):
+        for record in self:
+            for product in record.product_ids:
+                product.image_1920 = self._get_pic_from_google(product.barcode)
+
+    def _get_pic_from_google(self, barcode):
+        """ Get the first pictures corresponding to the search of the barcode on Google Image
+
+        Using Google Custom Search API, search for an image corresponding to the product
+        on Google using the Barcode as query.
+        If there is at least one result, take the first image and return it.
+
+        :param barcode: Barcode
+        :type barcode: Char
+        :return: The first image from the search or False if no image is found
+        :rtype: base64 image
+        """
+        payload = {
+            'cx': self.env['ir.config_parameter'].get_param('google.custom_search.cx').strip(),
+            'q': barcode,
+            'safe': 'active',
+            'searchType': 'image',
+            'key': self.env['ir.config_parameter'].get_param('google.custom_search.key').strip(),
+            'rights': 'cc_publicdomain,cc_attribute,cc_sharealike',
+            'imgsize': 'large',
+            'num': 1,
+        }
+        try:
+            response = requests.get('https://customsearch.googleapis.com/customsearch/v1',
+                params=payload, timeout=5)
+            response.raise_for_status()
+        except HTTPError as exception:
+            if exception.response.status_code == 429:
+                raise UserError(_("You reach your daily request limit included in your plan."))
+            elif exception.response.status_code == 403:
+                project_url = re.search(r"(?P<url>https?://[^\s]+)",
+                                exception.response.json().get('error').get('message')).group("url")
+                raise UserError(_("Custom Search API is not enabled in your Google project. \
+Enable it by visiting %s then retry. If you enabled this API recently, wait a few \
+minutes for the action to propagate to Google's systems and retry." % project_url))
+            elif exception.response.status_code == 400:
+                raise UserError(_("Your API Key or your Search Engine ID is incorrect."))
+            else:
+                raise
+        except Timeout as exception:
+            raise UserError(_("The Google Images API took too long to respond. \
+Please try again later."))
```

Comment:
Just use implicit string concatenation if you want to avoid long lines, it avoids messing up with the left margin:
```suggestion
            raise UserError(_("The Google Images API took too long to respond. "
                              "Please try again later."))
```